src/hooks/search-params-state.ts

```diff
@@ -0,0 +1,64 @@
+import { useMemo } from "react";
+import { useRouter } from "next/router";
+
+import useConst from "@/hooks/const";
+import useMemoizedFn from "@/hooks/memorized-fn";
+import { paramsSerializer } from "@/lib/base-http-client";
+
+/**
+ * useSearchParamsState
+ * - Provide state and setState for search params
+ * @param initialState a const variable, its change will not trigger re-render
+ * @returns [state, setState]
+ */
+export default function useSearchParamsState<
```

Comment:
Will this be used on the events page to handle the query filters instead?

Reply:
Already using in event page